### PR TITLE
Improve logic to show entities saved panel description.

### LINKED ISSUE #67963: Save panel: Improve logic to display conditional description
### Description

https://github.com/WordPress/gutenberg/pull/57716 added a descriptive text `Select the items you want to save.` to the entities saved panel. To my understanding, this message is shown when no changes are checked in the list of changes. That makes sense.

However, in the Theme Preview this message may be shown when there are no changes to check. In this case, the message is confusing and should be hidden.

The entities saved panel in the right sidebar and in the save modal dialog:

Image

It's also worth mentioning that, when there are changes listed in the panel, this message only appears when _unchecking all_ the checkboxes. I'm not sure that's the intended behavior from the original PR and I'm wondering whether this message should be removed entirely.

Cc @t-hamano 

Animated GIF to illustrate the message appears only when unchecking all the checkboxes:

Image

### Step-by-step reproduction instructions

- Go to the WP admin > Appearance > any non-active block theme > Live Preview
- Click 'Activate {theme name}', whether it's the one in the navigation panel or the one at the top right of the screen doesn't matter.
- Observe the entities saved panel shows the message `Select the items you want to save.` but there's no items to check.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

### PR DESCRIPTION
**Pull request by afercia**
Fixes https://github.com/WordPress/gutenberg/issues/67963

## What?

The Theme Preview save panel shows a description `Select the items you want to save.` even when there are no change items to save.

## Why?

The message is confusing and the logic to show it should be improved.

## How?

Shows the message only when there are actual dirty entities records to save.

## Testing Instructions

- Repeat the steps from the issue https://github.com/WordPress/gutenberg/issues/67963
- Observe the description message is shown only when appropriate.

## Screenshots or screencast 

|Before|After|
|-|-|
|||

### CODE REVIEW

File: packages/editor/src/components/entities-saved-states/index.js [hunk 1]
```diff
@@ -115,6 +115,10 @@ export function EntitiesSavedStatesExtensible( {
 		'description'
 	);
 
+	const selectItemsToSaveDescription = dirtyEntityRecords.length
```
Comment (Mamaduka) [inline packages/editor/src/components/entities-saved-states/index.js]:
```suggestion
	const selectItemsToSaveDescription = !! dirtyEntityRecords.length
```

Let's use a boolean for the condition.

**Review by Mamaduka — APPROVED** [review]
I just left a minor note. The logic change works as expected ✅

**Comment by afercia** [comment]
@Mamaduka thanks for your review. Applied your suggestion and waiting for tests to pass before hitting the green button.

**Review by t-hamano — APPROVED** [review]
Thanks for the fix. Certainly, when we try to activate a theme that has not had any changes made to it, that text should not be displayed.

**Pull request merged**